### ISSUE TITLE
can't flash vbmeta, or can't boot GSI - unisoc device sc9863a

### ISSUE THREAD

**Issue by mccalas89** [phhusson/treble_experimentations#1602]
Hi, I have a Blu V70 (same hardware/software as G60 - device codename G0270WW) with a Unisoc sc9863 processor. I unlocked bootloader using a modified fastboot posted on hovatek.com and used the signidentifier token to do so. In order to root with Magisk, i had to patch the recovery.img (my ramdisk in recovery), create a custom signed vbmeta.img, and then flash both before device would accept.

I wanted to try running GSI rom (my device is Treble enabled, slot A only, system-as-root), but I have been unable to do so. I have followed the instructions from many different places (various XDA threads, google's GSI page), but I can't seem to get things to work.

1) my device will not accept any vbmeta.img that's not 1024 kB : after 'fastboot flash vbmeta vbmeta.img',  with or without --disable-verifty/verification, fastboot hangs on "writing vbmeta." This seems to be an issue with this vbmeta being unsigned or not being the expected size. Neither will it accept a stock vbmeta.img with "--disable-verity" or "--disable-verification". However, if i pad a blank vbmeta (or another that comes with google's android gsi-pie image, for example) to match the stock size, THEN i can flash vbmeta (and was able to --disable-verification).

2) without flashing vbmeta.img, the GSI system.img flashes, but bootloops (very quickly i might add, it briefly hangs for a second on boot, then reboots). with flashing a vbmeta (a blank padded one) the GSI system flashes, but bootloops again (slightly differently. it hangs for 5-10 seconds on boot, then reboots).

Does anyone have any suggestions? Ideas? I hope this is an ok place to ask, wasn't sure where to go. Thanks for any help.

**Comment by phhusson** [comment]
Which GSI variant have you flashed?

**Comment by mccalas89** [comment]
Wow, that was fast!

I flashed AOSP 9.0 v119,
arm64_a-only, gapps, su
arm64_a-only, vanillo, nosu

and for good measure, i tried a/b versions as well (someone on reddit with a similar device did this, same sc9863a on Blu Vivo X5: https://www.reddit.com/r/androidroot/comments/fpky0z/how_i_rooted_and_unlocked_the_bootloader_of_and/ )

**Comment by phhusson** [comment]
Yeah you need A/B on SaR.
Fwiw I have a Qin 2 Pro with that SoC that boots fine.
My aosp 9 is long dead. Can you try my aosp 10?

**Comment by mccalas89** [comment]
i'll download now, let you know in a bit

**Comment by mccalas89** [comment]
amazing, your aosp 10 booted. didn't need to flash vbmeta, just system. i'll check to see what works and what doesn't.

i was so sure that b/c my device came with android 9 i should use your aosp 9.0 rom...

anyway, thanks so much!

**Comment by Chandresh-Sharma** [comment]
Please Help..........
I accidentally deleted everything on my android( **unisoc device sc9863a**) to unlock **bootloader** of my device(**Gionee F9**) But now my phone is not starting it is not even Going in recovery mode( **it is Hard bricked**) I have tried many ways to unlock bootloader but everytime I got error you described. So I Think this is the right place to discuss And Also share **how to unlock bootloader** Because it's removed pls help waiting for your reply.

**Comment by bzntm** [comment]
> Please Help.......... I accidentally deleted everything on my android( **unisoc device sc9863a**) to unlock **bootloader** of my device(**Gionee F9**) But now my phone is not starting it is not even Going in recovery mode( **it is Hard bricked**) I have tried many ways to unlock bootloader but everytime I got error you described. So I Think this is the right place to discuss And Also share **how to unlock bootloader** Because it's removed pls help waiting for your reply.

I assume you don't enable OEM unlock, so you MUST flash the original firmware using ResearchDownload (or something similar), enable Developer Options, and then enable OEM unlock
After that, boot into Fastboot (you need to use modded Fastboot from Hovatek to send unlock command)

**Comment by gparmvir980** [comment]
Bro I got same (GIONEE F9) I try hovatek modded fast boot but at final step it's fail to unlock boot loader
Please help:)

**Comment by Chandresh-Sharma** [comment]
> Bro I got same (GIONEE F9) I try hovatek modded fast boot but at final step it's fail to unlock boot loader Please help:)

What do you want to do with Gionee F9

**Comment by gparmvir980** [comment]
Actually to root my device and flash ROMs or custom imgs

**Comment by vanelizarov** [comment]
@gparmvir980 got the same problem with Teclast T40 Pro 2023 Z3D1
Can't unlock bootloader with Hovatek method, seems like Unisoc private key is not working anymore

**Comment by tunermason** [comment]
They much have changed the production key

**Comment by Chandresh-Sharma** [comment]
> They much have changed the production key

Do you have any idea how to root Gionee F9

**Comment by tunermason** [comment]
> > They much have changed the production key
> 
> Do you have any idea how to root Gionee F9

Unlocking the bootloader would be the first step before doing anything. What is the chipset? unisoc SC9863A

**Comment by Chandresh-Sharma** [comment]
> > > They much have changed the production key
> > 
> > 
> > Do you have any idea how to root Gionee F9
> 
> Unlocking the bootloader would be the first step before doing anything. What is the chipset? unisoc SC9863A

Yes it unisoc SC9863A and I also know we should first unlock bootloader and that's the main issue

**Comment by tunermason** [comment]
Yeah, I'm kind of dead in the water as well. Without the proper signing key, the firmware won't accept the unlock commands. Unless someone was to exploit the physical security of the chip and successfully gain full access there is not much we can do. The key would need to be leaked or the physical exploit published. Even if you are able to gain control of one chip the key wouldn't be stored on the chip but most likely a hash. The firmware will return false errors such as "command not found" and similar responses if there is not a signed command. For now, Chicom remains in control of the chips... The key could be changed for the white labeler or even production batches, this is the issue with closed access hardware, you don't fully own or control your own property.

**Comment by tunermason** [comment]
There was an exploit that hit the web a couple of months ago but all most all of the vulnerabilities were patched with an over the air update from the device manufacturers. With the hardware control mechanisms in place, it also makes it nearly impossible to remove spyware or other back doors installed in the OS.
